vllm/model_executor/layers/quantization/compressed_tensors/schemes/compressed_tensors_w4a4_nvfp4.py

```diff
@@ -95,8 +95,8 @@ def swizzle_blockscale(self, scale: torch.tensor):
                                             cols // 4, 4)
         swizzled_scale = padded_scale.permute((0, 1, 4, 3, 2, 5))
         swizzled_scale = swizzled_scale.contiguous().cuda()
-        return (swizzled_scale.reshape(M, K)
-                if scale_ndim == 2 else swizzled_scale.reshape(B, M, K))
+        return (swizzled_scale.reshape(M_padded, K_padded) if scale_ndim == 2
+                else swizzled_scale.reshape(B, M_padded, K_padded))
```

Comment:
Could you look at the shared utility and fix there if needed? https://github.com/vllm-project/vllm/blob/c6cd5ca3d3294f62bf5fad25ada8192ba39249b9/vllm/model_executor/layers/quantization/utils/quant_utils.py#L599